modules/videopress/class.videopress-gutenberg.php

```diff
@@ -46,17 +86,21 @@ public static function register_video_block_with_videopress() {
 	 *
 	 * @return string
 	 */
-	public static function render_video_block_with_videopress( $attributes, $content ) {
+	public function render_video_block_with_videopress( $attributes, $content ) {
 		if ( ! isset( $attributes['id'] ) || isset( $attributes['guid'] ) ) {
 			return $content;
 		}
 
 		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
 			$blog_id = get_current_blog_id();
-		} else {
+		} elseif ( method_exists( 'Jetpack', 'is_active' ) && Jetpack::is_active() ) {
 			$blog_id = Jetpack_Options::get_option( 'id' );
 		}
 
+		if ( ! isset( $blog_id ) ) {
+			return $content;
+		}
+
```

Comment:
I'm a bit confused about L94-L103; shouldn't `get_current_blog_id()` work for both WP.com and JP alike?

(Picking the right blog ID seems like such a universal problem that I'd hope we have code that addresses it generically -- otherwise, pretty much every file in JP would have to include logic like the above, wouldn't it? Which I'd find somewhat hard to believe...)

Reply:
I always got the correct id when using `get_current_blog_id` but @dmsnell was getting unexpected values which were solved when using `Jetpack_Options::get_option( 'id' )`  ([#](https://github.com/Automattic/jetpack/pull/11193#issuecomment-457883886), [#](https://github.com/Automattic/jetpack/pull/11193/commits/215cf789f3d8bd03ff9eb1bbdb693acb8831d273)).

Reply:
Hmm, I've seen some precedent where -- in code invoked through a REST API endpoint -- `get_current_blog_id()` would return `public-api.wordpress.com`'s site ID. This happened most likely since it was called through a hook that was triggered before the relevant `switch_to_blog()` call, and could be fixed by invoking the relevant action from the function that was called to provide the endpoint's response.

I don't think I've personally seen `get_current_blog_id()` return `1` yet :thinking: I wonder if that could be prevented by setting a later priority to the `add_action` call that attaches the enclosing function to `init`. Not that guessing priorities is exactly elegant, but it's arguably kinda WordPress-y :stuck_out_tongue: 

Anyway, if you keep it the way it is now, would you mind adding an inline comment to the code explaining why we're not using `get_current_blog_id()` in the Jetpack case? (i.e. essentially what you explained in https://github.com/Automattic/jetpack/pull/11321#discussion_r255600102.)

Reply:
Done in 117d95b